src/lib.rs

```diff
@@ -63,170 +40,134 @@ pub enum Error {
     /// No space for the region
     OutOfSpace,
 
-    /// Not inside the address space
-    Overflow,
-
-    /// Overlap with the existing regions
+    /// Overlapping with the existing regions
     Overlap,
 
-    /// Invalid region
+    /// Invalid region given as input
     InvalidRegion,
 }
 
 /// A virtual memory map ledger.
-//
-// Developer Note: the first record is reserved for the ledger bounds. We
-// structure it this way so that the user of the `Ledger` type has
-// fine-grained controls over allocation. For example, to allocate a 4k page,
-// the user can instantiate as `Ledger::<128>::new(..)`.
 #[derive(Clone, Debug)]
-pub struct Ledger<const N: usize> {
-    records: [Record; N],
+pub struct Ledger<V, const N: usize> {
+    limits: Line<Address<usize, Page>>,
+    regions: [Region<V>; N],
+    len: usize,
```

Comment:
Having multiple fields in the ledger means we lose fine-grained control of memory allocation for the ledger. In the previous version, it would be easy to make it page aligned. We have no hope of controlling that now. I designed the data types the way I did for a reason.

Reply:
IMHO, it was quite convoluted, especially the length field. I can fix the alignment of course.